sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -853,6 +853,35 @@ Flux<ServiceBusMessageContext> receiveMessagesWithContext(int highTide) {
             .onErrorMap(throwable -> mapError(throwable, ServiceBusErrorSource.RECEIVE));
     }
 
+    /**
+     * Receives streams of {@link ServiceBusReceivedMessage messages} from rolling sessions.
+     * This method only support for session processor to receive message, so receive option must
+     * be the rolling session receiver.
+     *
+     * @return A stream of session which can receive session messages.
+     *
+     * @throws RuntimeException if receiver is not a rolling session receiver.
+     */
+    Flux<Flux<ServiceBusMessageContext>> receiveMessagesWithContextFromRollingSessions() {
+        if (receiverOptions.isRollingSessionReceiver()) {
```

Comment:
Do we need to check this here, I mean we already check receiver type before.

Reply:
I am afraid this function is called from other places.

Reply:
What is the major change in this PR? Lots of the code is refactor, so it is hard to find what matters.

Reply:
I add ` isRollingSessionReceiver()` check in processor to treat receiving session message separately. The change for this PR includes:
- Add new method `receiveFromRollingSessions()` to return the Flux of <each session messages> rather than the merged messages. If we use session processor, now it goes into `receiver.receiveFromRollingSessions()`, and previous it used `receiver.receive()`.
- Wraps message subscriber definition in function `createMessageSubscriber()` so that we can reuse code for both session and non-session processor.